gammapy/irf/edisp/core.py

```diff
@@ -117,7 +124,13 @@ def from_gauss(
             data=data.value,
         )
 
-    def to_edisp_kernel(self, offset, energy_true=None, energy=None):
+    @deprecated_renamed_argument(
+        ["energy_true", "energy"],
+        ["energy_axis_true", "energy_axis"],
```

Comment:
Do you want to rename the argument name or keep the name and change the type?
Either of the two should suffice, no?

If a `DepricationWarning` is raised about the argument name, then the `warning` below is not necessary.

I think we can keep the argument same same, and just allow a `MapAxis` on input. Then, we can raise a deprecation warning if a Quantity is passed

Reply:
I changed the argument name to be consistent with others in the code, for example [here](https://github.com/gammapy/gammapy/blob/18b65d5cce21876d53aa7faf54aaeccd68e8b76c/gammapy/irf/edisp/map.py#L116).

Reply:
Okay sure thing! Will do

Reply:
So `deprecated_renamed_argument` will only let the user know that there was a name change, but not that it should now be a `MapAxis`